From ffde703470b03b1000017ed35c4f90a90caa22cf Mon Sep 17 00:00:00 2001
Message-Id: <ffde703470b03b1000017ed35c4f90a90caa22cf.1638342746.git.geliang.tang@suse.com>
From: Kees Cook <keescook@chromium.org>
Date: Tue, 27 Oct 2020 12:23:19 -0700
Subject: [PATCH] arm64: Enable seccomp architecture tracking
Git-commit: ffde703470b03b1000017ed35c4f90a90caa22cf
Patch-mainline: v5.11-rc1
References: bsn#22

To enable seccomp constant action bitmaps, we need to have a static
mapping to the audit architecture and system call table size. Add these
for arm64.

Signed-off-by: Kees Cook <keescook@chromium.org>
Signed-off-by: Geliang Tang <geliang.tang@suse.com>
---
 arch/arm64/include/asm/seccomp.h | 9 +++++++++
 1 file changed, 9 insertions(+)

diff --git a/arch/arm64/include/asm/seccomp.h b/arch/arm64/include/asm/seccomp.h
index c36387170936..30256233788b 100644
--- a/arch/arm64/include/asm/seccomp.h
+++ b/arch/arm64/include/asm/seccomp.h
@@ -19,4 +19,13 @@
 
 #include <asm-generic/seccomp.h>
 
+#define SECCOMP_ARCH_NATIVE		AUDIT_ARCH_AARCH64
+#define SECCOMP_ARCH_NATIVE_NR		NR_syscalls
+#define SECCOMP_ARCH_NATIVE_NAME	"aarch64"
+#ifdef CONFIG_COMPAT
+# define SECCOMP_ARCH_COMPAT		AUDIT_ARCH_ARM
+# define SECCOMP_ARCH_COMPAT_NR	__NR_compat_syscalls
+# define SECCOMP_ARCH_COMPAT_NAME	"arm"
+#endif
+
 #endif /* _ASM_SECCOMP_H */
-- 
2.31.1

